Assets/Scripts/GameTime/Clock.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using Managers;
+using UnityEngine;
+
+namespace GameTime
+{
+    public class Clock : MonoBehaviour
+    {
+        [SerializeField][Tooltip("Day length in minutes")]
+        private float dayLength;
+    
+        [SerializeField][Tooltip("Year length in days")]
+        private int yearLength;
+
+        [SerializeField] [Tooltip("Current time")] [Range(0f, 1f)]
+        private float timeOfDay;
+        public float TimeOfDay => timeOfDay;
+
+        private int _days;
+        public int Days => _days;
+
+        private int _years;
+        public int Years => _years;
+        
+        private float _timeScale;
+        
+        [SerializeField][Tooltip("Controls timeScale so certain time periods (night) can pass faster than others (day)")]
+        private AnimationCurve timeCurve;
+
+        private float _timeCurveNormalization;
+
+        private void Awake()
+        {
+            NormalizeTimeCurve();
+        }
+
+        private void Update()
+        {
+            UpdateTimeScale();
+            UpdateTime();
+        }
+
+        /// <summary>
+        /// Updates the current timeScale; Varies depending on the current time of day (=> timeCurve)
+        /// </summary>
+        private void UpdateTimeScale()
+        {
+            _timeScale = 24 / (dayLength / 60);
```

Comment:
seems like this calculation is constant, so maybe move that out of the update method to save resources

Reply:
good idea

Reply:
you moved it from UpdateTimeScale to Update? 😀